server/rpc/interceptors/interceptors.go

```diff
@@ -54,6 +57,12 @@ func ClientOrigin() string {
 	return *origin
 }
 
+// PropagateLabelsContextKey is a context key that, if present in the context,
+// causes the configured origin and client headers to propagate to the outgoing
+// context. These headers will also be propagated as-is across chained RPCs to
+// other BuildBuddy servers if applicable.
+type PropagateLabelsContextKey struct{}
```

Comment:
Do we need this context key? Or would it be sufficient to run:

```
		if *origin != "" {
			ctx = metadata.AppendToOutgoingContext(ctx, "x-buildbuddy-origin", *origin)
		}
		if *client != "" {
			ctx = metadata.AppendToOutgoingContext(ctx, "x-buildbuddy-client", *client)
		}
```

when InternalUsageContext() is called?

Reply:
Good call :) done